corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -3,43 +3,66 @@ hqDefine('data_dictionary/js/data_dictionary.js', function () {
         var self = this;
-        self.name = ko.observable(name);
+        self.name = name;
         self.properties = ko.observableArray();
 
         self.init = function (group_dict, changeSaveButton) {
             _.each(group_dict, function (properties, group) {
                 var groupObj = new PropertyListItem(group, true, group, self.name);
                 self.properties.push(groupObj);
                 _.each(properties, function (prop) {
-                    var propObj = new PropertyListItem(prop.name, false, prop.group, self.name, prop.data_type, prop.description);
+                    var propObj = new PropertyListItem(prop.name, false, prop.group, self.name, prop.data_type, prop.description, prop.deprecated);
                     propObj.description.subscribe(changeSaveButton);
                     propObj.dataType.subscribe(changeSaveButton);
+                    propObj.deprecated.subscribe(changeSaveButton);
                     self.properties.push(propObj);
                 });
             });
         };
     };
 
-    var PropertyListItem = function (name, isGroup, groupName, caseType, dataType, description) {
+    var PropertyListItem = function (name, isGroup, groupName, caseType, dataType, description, deprecated) {
         var self = this;
-        self.name = ko.observable(name);
+        self.name = name;
         self.expanded = ko.observable(true);
         self.isGroup = isGroup;
         self.group = ko.observable(groupName);
         self.caseType = caseType;
         self.dataType = ko.observable(dataType);
         self.description = ko.observable(description);
+        self.deprecated = ko.observable(deprecated || false);
         self.availableDataTypes = ko.observableArray([
             {value: 'date', display: django.gettext('Date')},
             {value: 'plain', display: django.gettext('Plain')},
             {value: 'number', display: django.gettext('Number')},
             {value: 'select', display: django.gettext('Select')},
             {value: 'integer', display: django.gettext('Integer')},
+            {value: 'audio', display: django.gettext('Audio')},
+            {value: 'barcode', display: django.gettext('Barcode')},
+            {value: 'gps', display: django.gettext('GPS')},
+            {value: 'image', display: django.gettext('Image')},
+            {value: 'phone_number', display: django.gettext('Phone Number')},
+            {value: 'password', display: django.gettext('Password')},
+            {value: 'signature', display: django.gettext('Signature')},
+            {value: 'video', display: django.gettext('Video')},
+            {value: 'balance', display: django.gettext('Balance')},
+            {value: 'transfer', display: django.gettext('Transfer')},
+            {value: 'dispense', display: django.gettext('Dispense')},
+            {value: 'receive', display: django.gettext('Receive')},
+            {value: 'save_to_case', display: django.gettext('Save To Case')},
```

Comment:
Not all of these are relevant to case properties e.g. `save_to_case`, `receive`, `dispense`, `transfer`, `balance`, `video`, `signature`, `image`, `audio`